cpp/include/cudf/contiguous_split.hpp

```diff
@@ -0,0 +1,229 @@
+ *
+ * For all `i` it is expected `splits[i] <= splits[i+1] <= input.size()`
+ * For a `splits` size N, there will always be N+1 splits in the output
+ *
+ * @note It is the caller's responsibility to ensure that the returned views
+ * do not outlive the viewed device memory contained in the `all_data` field of the
+ * returned packed_table.
+ *
+ * @code{.pseudo}
+ * Example:
+ * input:   [{10, 12, 14, 16, 18, 20, 22, 24, 26, 28},
+ *           {50, 52, 54, 56, 58, 60, 62, 64, 66, 68}]
+ * splits:  {2, 5, 9}
+ * output:  [{{10, 12}, {14, 16, 18}, {20, 22, 24, 26}, {28}},
+ *           {{50, 52}, {54, 56, 58}, {60, 62, 64, 66}, {68}}]
+ * @endcode
+ *
+ *
+ * @throws cudf::logic_error if `splits` has end index > size of `input`.
+ * @throws cudf::logic_error When the value in `splits` is not in the range [0, input.size()).
+ * @throws cudf::logic_error When the values in the `splits` are 'strictly decreasing'.
+ *
+ * @param input View of a table to split
+ * @param splits A vector of indices where the view will be split
+ * @param[in] mr Device memory resource used to allocate the returned result's device memory
+ * @return The set of requested views of `input` indicated by the `splits` and the viewed memory
+ * buffer.
+ */
+std::vector<packed_table> contiguous_split(
+  cudf::table_view const& input,
+  std::vector<size_type> const& splits,
+  rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource());
+
+/**
+ * @brief Deep-copy a `table_view` into a serialized contiguous memory format
+ *
+ * The metadata from the `table_view` is copied into a host vector of bytes and the data from the
+ * `table_view` is copied into a `device_buffer`. Pass the output of this function into
+ * `cudf::unpack` to deserialize.
+ *
+ * @param input View of the table to pack
+ * @param[in] mr Optional, The resource to use for all returned device allocations
+ * @return packed_columns A struct containing the serialized metadata and data in contiguous host
+ *         and device memory respectively
+ */
+packed_columns pack(cudf::table_view const& input,
+                    rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource());
+
+/**
+ * @brief Produce the metadata used for packing a table stored in a contiguous buffer.
+ *
+ * The metadata from the `table_view` is copied into a host vector of bytes which can be used to
+ * construct a `packed_columns` or `packed_table` structure.  The caller is responsible for
+ * guaranteeing that that all of the columns in the table point into `contiguous_buffer`.
+ *
+ * @param table View of the table to pack
+ * @param contiguous_buffer A contiguous buffer of device memory which contains the data referenced
+ * by the columns in `table`
+ * @param buffer_size The size of `contiguous_buffer`
+ * @return Vector of bytes representing the metadata used to `unpack` a packed_columns struct
+ */
+packed_columns::metadata pack_metadata(table_view const& table,
```

Comment:
Similar to my comment above, just use `std::vector<uint8_t/std::byte>` directly.